ansible/doc/README.testbed.VsSetup.md

```diff
@@ -1,77 +1,100 @@
 ```
-#### Option 2, download and image cEOS image automatically
-Alternatively, you can host the cEOS image on a http server. Specify `vm_images_url` for downloading the image [here](https://github.com/Azure/sonic-mgmt/blob/master/ansible/group_vars/vm_host/main.yml#L2). If a saskey is required for downloading cEOS image, specify `ceosimage_saskey` in `sonic-mgmt/ansible/vars/azure_storage.yml`.
 
-If you want to skip image downloading when the cEOS image is not imported locally, set `skip_ceos_image_downloading` to `true` in `sonic-mgmt/ansible/group_vars/all/ceos.yml`. Then when cEOS image is not locally imported, the scripts will not try to download it and will fail with an error message. Please use option 1 to download and import the cEOS image manually.
+#### Option 2.2: Pull cEOS image automatically
+1. Alternatively, you can host the cEOS image on a http server. Specify `vm_images_url` for downloading the image [here](https://github.com/Azure/sonic-mgmt/blob/master/ansible/group_vars/vm_host/main.yml#L2).
+
+2. If a SAS key is required for downloading the cEOS image, specify `ceosimage_saskey` in `sonic-mgmt/ansible/vars/azure_storage.yml`.
+
+If you want to skip downloading the image when the cEOS image is not imported locally, set `skip_ceos_image_downloading` to `true` in `sonic-mgmt/ansible/group_vars/all/ceos.yml`. Then, when the cEOS image is not locally available, the scripts will not try to download it and will fail with an error message. Please use option 1 to download and import the cEOS image manually.
 
-## Download sonic-vs image
+## Download the sonic-vs image
+To run the tests with a virtual SONiC device, we need a virtual SONiC image. The simplest way to do so is to download a public build from Jenkins.
+
+1. Download the sonic-vs image from [here](https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-image/lastSuccessfulBuild/artifact/target/sonic-vs.img.gz)
 
-- Download sonic-vs image from [here](https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-image/lastSuccessfulBuild/artifact/target/sonic-vs.img.gz)
 ```
 $ wget https://sonic-jenkins.westus2.cloudapp.azure.com/job/vs/job/buildimage-vs-image/lastSuccessfulBuild/artifact/target/sonic-vs.img.gz
 ```
 
-- unzip the image and move it into `~/sonic-vm/images/`
+2. Unzip the image and move it into `~/sonic-vm/images/`
+
 ```
 $ gzip -d sonic-vs.img.gz
 $ mkdir -p ~/sonic-vm/images
 $ mv sonic-vs.img ~/sonic-vm/images
 ```
 
 ## Setup sonic-mgmt docker
+All testbed configuration steps and tests are run from a `sonic-mgmt` docker container. This container has all the necessary packages and tools for SONiC testing so that test behavior is consistent between different developers and lab setups.
 
-Run the `setup-container.sh` in the root directory of the sonic-mgmt repository:
+1. Run the `setup-container.sh` in the root directory of the sonic-mgmt repository:
```

Comment:
there are some pre-requiste to run setup-container.sh, basically j2cli and docker-engine. can you add those dependency in the doc as well?"

Reply:
Good call, added!